src/lapack/backends/cusolver/cusolver_helper.hpp

```diff
@@ -274,26 +280,43 @@ struct CudaEquivalentType<std::complex<double>> {
 
 /* devinfo */
 
-inline int get_cusolver_devinfo(sycl::queue &queue, sycl::buffer<int> &devInfo) {
-    sycl::host_accessor<int, 1, sycl::access::mode::read> dev_info_{ devInfo };
-    return dev_info_[0];
+inline void get_cusolver_devinfo(sycl::queue &queue, sycl::buffer<int> &devInfo,
+                                 std::vector<int> &dev_info_) {
+    sycl::host_accessor<int, 1, sycl::access::mode::read> dev_info_acc{ devInfo };
+    for (unsigned int i = 0; i < dev_info_.size(); ++i)
+        dev_info_[i] = dev_info_acc[i];
 }
 
-inline int get_cusolver_devinfo(sycl::queue &queue, const int *devInfo) {
-    int dev_info_;
+inline void get_cusolver_devinfo(sycl::queue &queue, const int *devInfo,
+                                 std::vector<int> &dev_info_) {
     queue.wait();
-    queue.memcpy(&dev_info_, devInfo, sizeof(int));
-    return dev_info_;
+    queue.memcpy(dev_info_.data(), devInfo, sizeof(int));
 }
 
 template <typename DEVINFO_T>
 inline void lapack_info_check(sycl::queue &queue, DEVINFO_T devinfo, const char *func_name,
-                              const char *cufunc_name) {
-    const int devinfo_ = get_cusolver_devinfo(queue, devinfo);
-    if (devinfo_ > 0)
-        throw oneapi::mkl::lapack::computation_error(
-            func_name, std::string(cufunc_name) + " failed with info = " + std::to_string(devinfo_),
-            devinfo_);
+                              const char *cufunc_name, int dev_info_size = 1) {
+    std::vector<int> dev_info_(dev_info_size);
+    get_cusolver_devinfo(queue, devinfo, dev_info_);
+    for (const auto &val : dev_info_) {
+        if (val > 0)
+            throw oneapi::mkl::lapack::computation_error(
+                func_name, std::string(cufunc_name) + " failed with info = " + std::to_string(val),
+                val);
+    }
+}
+
+/* batched helpers */
+
+// Creates list of matrix/vector pointers from initial ptr and stride
+// Note: user is responsible for deallocating memory
```

Comment:
Looks like none of the usages follow this?

Reply:
Thanks for catching this. I have updated this to free all the instances where this allocates memory